### PR TITLE
systemctl-edit fixes

### PR DESCRIPTION
**Pull request by anitazha**
Details in the commits. Fixes an abort and what I think is a missed argument.

### CODE REVIEW

File: src/systemctl/systemctl-edit.c [hunk 1]
```diff
@@ -398,14 +398,14 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
...
+                        r = unit_find_paths(bus, *name, &lp, true, &cached_name_map, &cached_id_map, &path, &unit_paths);
                 }
                 if (r == -ERFKILL)
                         return log_error_errno(r, "Unit '%s' masked, cannot edit.", *name);
                 if (r < 0)
                         return r;
 
-                if (r == 0) {
+                if (r == 0 || !path) {
                         assert(!path);
```
Comment (bluca) [inline src/systemctl/systemctl-edit.c]: you can remove this assert now

Comment (bluca) [reply src/systemctl/systemctl-edit.c]: ah no sorry, ignore me - there's the r == 0 case - need to remember to drink my morning tea before doing code reviews

**Review by bluca — APPROVED** [review]
lgtm

File: src/systemctl/systemctl-edit.c [hunk 2]
```diff
@@ -405,7 +405,7 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
...
```
Comment (poettering) [inline src/systemctl/systemctl-edit.c]: umm, this conditionalization in combination with the assert is weird. the assert says path is definitely NULL here. and that means the r == 0 test is entirely redundant?

**Review by poettering — CHANGES_REQUESTED** [review]

File: src/systemctl/systemctl-edit.c (hunk 2 above)
Comment (anitazha) [reply src/systemctl/systemctl-edit.c]: Uhhh yes though I had left in the `assert` in the event that `r == 0` somehow didn't end up with NULL path (as in the `else` case below where `r != 0` ended up with NULL path). I guess neither of that matters if I only check for `!path` and ignore `r`

**Pull request merged**